### PR TITLE
Simplify work packets

### PR DESCRIPTION
**Pull request by wenyuzhao**
Remove the `Generic Scheduler` <-> `MMTk Scheduler` abstraction. We don't really need to have the generic work packets abstraction. This simplifies the code a lot.

### CODE REVIEW

File: src/scheduler/worker.rs [hunk 1]
```diff
@@ -3,30 +3,42 @@ use super::work_bucket::*;
...
 
+/// Thread-local data for each worker thread.
+///
+/// For mmtk, each gc can define their own worker-local data, to contain their required copy allocators and other stuffs.
+pub trait GCWorkerLocal {
+    fn init(&mut self, _tls: VMWorkerThread) {}
+}
+
+/// A default implementation for scheduling systems that does not require a worker-local context.
+impl GCWorkerLocal for () {}
```
Comment (qinsoon) [inline src/scheduler/worker.rs:20]: Do we still need this?

**Review by qinsoon — APPROVED** [review]
Looks good. The scheduler type is used in the API. So you might need to update the bindings. I started the checks for bindings. If anything fails, please update the bindings.

**Comment by wenyuzhao** [comment]
> Looks good. The scheduler type is used in the API. So you might need to update the bindings. I started the checks for bindings. If anything fails, please update the bindings.

The scheduler type visibility is `pub(crate)`. So looks like bindings do not need to be updated.

File: src/scheduler/worker.rs (hunk 1 above)
Comment (wenyuzhao) [reply src/scheduler/worker.rs]: Thanks for pointing this out. We don't need this anymore. I've removed it.

**Pull request merged**